### PR TITLE
 MBS-10101 / MBS-9082: SecondHandSongs cleanup and autoselect additions

### PR DESCRIPTION
**Pull request by reosarevok**
https://tickets.metabrainz.org/browse/MBS-10101
https://tickets.metabrainz.org/browse/MBS-9082

I still suck at regexes, so please take a good look at those :)

### CODE REVIEW

File: root/static/scripts/edit/URLCleanup.js [hunk 1]
```diff
@@ -1818,6 +1818,11 @@ const CLEANUPS = {
   'secondhandsongs': {
     match: [new RegExp('^(https?://)?([^/]+\\.)?secondhandsongs\\.com/', 'i')],
     type: LINK_TYPES.secondhandsongs,
+    clean: function (url) {
+      url = url.replace(/^(https?:\/\/)?([^/]+\.)?secondhandsongs\.com/, 'https://secondhandsongs.com');
```
Comment (yvanzo) [inline root/static/scripts/edit/URLCleanup.js]: Capturing groups `(…)` in this regular expression are not used in the replacement string, thus these should be replaced with non-capturing groups `(?:…)`.

File: root/static/scripts/edit/URLCleanup.js [hunk 2]
```diff
@@ -1818,6 +1818,11 @@ const CLEANUPS = {
...
+      url = url.replace(/secondhandsongs\.com\/(\w+)\/([\d+]+)(?:[\/#?-].*)$/, 'secondhandsongs.com/$1/$2');
```
Comment (yvanzo) [inline root/static/scripts/edit/URLCleanup.js]: * URL should probably match from the beginning using `^https:\/\/`
* A single capturing group could be used for the whole not trimmed part and just have `'$1'` for the replacement string

File: root/static/scripts/tests/Control/URLCleanup.js [hunk 3]
```diff
@@ -2603,33 +2603,38 @@ const testData = [
...
-                     input_url: 'http://www.secondhandsongs.com/release/888',
+                     input_url: 'http://secondhandsongs.com/release/888',
              input_entity_type: 'release',
     expected_relationship_type: 'secondhandsongs',
+            expected_clean_url: 'https://secondhandsongs.com/release/888',
        only_valid_entity_types: ['release'],
   },
   {
-                     input_url: 'http://www.secondhandsongs.com/work/1409',
+                     input_url: 'https://secondhandsongs.com/work/140348/adaptations#nav-entity',
```
Comment (yvanzo) [inline root/static/scripts/tests/Control/URLCleanup.js]: Changing scheme `http` and authority `www.…` parts is weakening test here.
If more test variations are needed, `label` and `performance` tests can be changed instead.

Comment (reosarevok) [reply root/static/scripts/tests/Control/URLCleanup.js]: I thought if this was tested once (on the artist one) we didn't need to test it more, but I guess it can't hurt. Will change.

Comment (yvanzo) [reply root/static/scripts/tests/Control/URLCleanup.js]: Not changing it should allow to test cleaning a combination of unclean scheme/authority part(s) and unclean path/query/fragment part(s).

**Comment by reosarevok** [comment]
@brainzbot, retest this please

File: root/static/scripts/edit/URLCleanup.js [hunk 4]
```diff
@@ -1816,8 +1816,13 @@ const CLEANUPS = {
     },
   },
   'secondhandsongs': {
-    match: [new RegExp('^(https?://)?([^/]+\\.)?secondhandsongs\\.com/', 'i')],
+    match: [new RegExp('^(https?://)?(?:[^/]+\\.)?secondhandsongs\\.com/', 'i')],
```
Comment (yvanzo) [inline root/static/scripts/edit/URLCleanup.js]: If you go this direction, none of the groups in `match` rules needs to be capturing, but that should probably be done in a separate commit.

File: root/static/scripts/edit/URLCleanup.js [hunk 5]
```diff
@@ -1816,8 +1816,13 @@ const CLEANUPS = {
...
     type: LINK_TYPES.secondhandsongs,
+    clean: function (url) {
+      url = url.replace(/^(?:https?:\/\/)?(?:[^/]+\.)?secondhandsongs\.com/, 'https://secondhandsongs.com');
+      url = url.replace(/^https:\/\/secondhandsongs\.com\/(\w+\/[\d+]+)[\/#?-].*$/, 'https://secondhandsongs.com/$1');
```
Comment (yvanzo) [inline root/static/scripts/edit/URLCleanup.js]: Capturing group could start with `https`… so that replacement string just needs to be `'$1'`.
